### PR TITLE
DSG-1565/fix-timestmap-audit-event-when-updating-documents

### PR DESCRIPTION
**Pull request by afcalderonm**
## Screenshots (if appropriate):
image

## Types of changes

Patch
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java [hunk 1]
```diff
@@ -815,7 +815,7 @@ public AuditEventId postAuditEventForDocumentStatusChanged(
                 document.getCreatedBy(),
                 document.getCreatedBy(),
                 summary,
-                document.getLastUpdatedTimestamp(),
+                document.getReviewedOn(),
```
Comment (JayStGelais) [inline service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:818]: This must use `RequestContextTimestamp.getCurrentTimestamp()`.

**Review by JayStGelais — CHANGES_REQUESTED** [review]

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java (hunk 1 above)
Comment (afcalderonm) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:818]: before sending the event the ReviewedOn is set with `RequestContextTimestamp.getCurrentTimestamp()` should i call here too?

Comment (JayStGelais) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:818]: Yes, lets always use the value from `RequestContextTimestamp` for all audit events. If we ever change the logic for updating `reviewedOn` in the future we wont break this timestamp.

Comment (asultan001) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:818]: Wait, I think that is a potential disconnect. Ideally we want to show the same times when you are looking the the audit event and the last reviewed on time so there may be value in using the exact same time and not something that may be off by a few milliseconds.

Comment (JayStGelais) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:818]: The version that first appeared in this PR follows this flow:

`RequestContextTimestamp.getCurrentTimestamp()` -> `reviewedOn` -> Event Timestamp

I'm proposing we switch to:

`RequestContextTimestamp.getCurrentTimestamp()` -> `reviewedOn` 
`RequestContextTimestamp.getCurrentTimestamp()` ->  Event Timestamp

They will still have the exact same timestamp, the one being held in `RequestContextTimestamp`. But now if we move the update to `reviewedOn`  into a JPA lifecycle hook, like we have done for similar timestamps in other entities, we wont introduce a bug where the Event Timestamp is based on old data.

Comment (afcalderonm) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java]: i just change it to use the `RequestContextTimestamp.getCurrentTimestamp() `

**Review by asultan001 — APPROVED** [review]
LGTM

**Review by JayStGelais — APPROVED** [review]

**Pull request merged**